storage_controller/src/http.rs

```diff
@@ -749,6 +750,55 @@ impl From<ReconcileError> for ApiError {
     }
 }
 
+/// Return the safekeeper record by instance id, or 404.
+///
+/// Not used by anything except manual testing.
+async fn handle_get_safekeeper(req: Request<Body>) -> Result<Response<Body>, ApiError> {
+    // TODO: jwt, see POST handler in handle_upsert_safekeeper
+    let id = parse_request_param::<i64>(&req, "id")?;
+
+    let state = get_state(&req);
+
+    let res = state.service.get_safekeeper(id).await;
+
+    match res {
+        Ok(b) => json_response(StatusCode::OK, b),
+        Err(crate::persistence::DatabaseError::Query(diesel::result::Error::NotFound)) => {
+            Err(ApiError::NotFound("unknown instance_id".into()))
+        }
+        Err(other) => Err(other.into()),
+    }
+}
+
+/// Used as part of deployment scripts.
+///
+/// Assumes information is only relayed to storage controller after first selecting an unique id on
+/// control plane database, which means we have an id field in the request and payload.
+async fn handle_upsert_safekeeper(mut req: Request<Body>) -> Result<Response<Body>, ApiError> {
+    // todo: jwt -- as long we don't use this for anything, it doesn't really need much security
+    // either
```

Comment:
I think you can add the token now. The `neondatabase/infra` already has a `infra_to_storage_controller_token` which is used by the `restart_storage_controller_pageserver_script.py`. It's `Admin` scoped, so you can just use that.

Reply:
done with [c60dcb3](https://github.com/neondatabase/neon/pull/8879/commits/c60dcb379e123026a629987169c32621223ced7c)